extra_scripts/create_sun397_data_files.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import argparse
+import dataclasses
+import os
+import random
+from typing import List
+
+import numpy as np
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+
+def get_argument_parser():
+    """
+    List of arguments supported by the script
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--input",
+        type=str,
+        help="Path to the folder containing the original SUN397 folder",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=str,
+        help="Folder where the classification dataset will be written",
+    )
+    parser.add_argument(
+        "-d",
+        "--download",
+        action="store_const",
+        const=True,
+        default=False,
+        help="To download the original dataset and decompress it in the input folder",
+    )
+    parser.add_argument(
+        "-s",
+        "--seed",
+        type=int,
+        default=0,
+        help="Random seed used to create the train/val/test splits",
+    )
+    return parser
+
+
+def download_dataset(root: str):
+    """
+    Download the SUN397 dataset archive and expand it in the folder provided as parameter
+    """
+    URL = "http://vision.princeton.edu/projects/2010/SUN/SUN397.tar.gz"
+    download_and_extract_archive(url=URL, download_root=root)
+
+
+TRAIN_SPLIT_SIZE = 0.7
+VALID_SPLIT_SIZE = 0.1
+
+
+@dataclasses.dataclass
+class SplitData:
+    """
+    Data structure holding the samples associated to a given split
+    """
+
+    image_paths: List[str] = dataclasses.field(default_factory=list)
+    image_labels: List[int] = dataclasses.field(default_factory=list)
+
+
+def split_sample_list(xs):
```

Comment:
nit: use a more descriptive input name instead of `xs` :)

Reply:
Oups... this "xs" came right out of my Haskell days ! I will rename this :)

Reply:
Fixed at a3f973b :)